src/lib/stepper/stepper.ts

```diff
@@ -15,26 +15,79 @@ import {
   // considers such imports as unused (https://github.com/Microsoft/TypeScript/issues/14953)
   // tslint:disable-next-line:no-unused-variable
   ElementRef,
+  Inject,
+  Optional,
   QueryList,
+  SkipSelf,
   ViewChildren
 }from '@angular/core';
 import {MdStepLabel} from './step-label';
+import {
+  defaultErrorStateMatcher,
+  ErrorOptions,
+  MD_ERROR_GLOBAL_OPTIONS,
+  ErrorStateMatcher
+} from '../core/error/error-options';
+import {
+  FormArray, FormControl, FormGroupDirective, NgForm, ValidationErrors,
+  ValidatorFn
+} from '@angular/forms';
+
+/**
+ * Form array validator to check if all form groups in form array are valid.
+ * If not, it will return the index of the first invalid form group.
+ */
+export const stepValidator: ValidatorFn = (formArray: FormArray): ValidationErrors | null => {
```

Comment:
should probably be called `stepperValidator` since it's attached to the `FormArray` representing the whole stepper, not the `FormGroup`s representing the steps.